ocean_lib/assets/utils.py

```diff
@@ -16,10 +16,10 @@ def create_checksum(text):
 
 
 @enforce_types_shim
-def generate_trusted_algo_dict(
-    did: str = None, metadata_cache_uri: str = None, ddo: Asset = None
-):
-    assert ddo or (did and metadata_cache_uri)
+def generate_trusted_algo_dict(did=None, metadata_cache_uri=None, ddo=None):
```

Comment:
why are you changing the parameters here?

Reply:
I have changed the parameters, because I needed to test with `None`, so when I deleted the type, the test worked fine, otherwise it failed with `NoneType for did, metadata_cache_uri and ddo` or something similar. What is your suggestion?

Reply:
https://stackoverflow.com/a/39429578/14214722
use Optional. @DMats has used it in the project before, so you most definitely can find samples.